packages/next/src/utils.ts

```diff
@@ -2377,6 +2379,49 @@ interface EdgeFunctionMatcher {
   originalSource?: string;
 }
 
+const vercelFunctionRegionsVar = process.env.VERCEL_FUNCTION_REGIONS;
+let vercelFunctionRegions: string[] | undefined;
+if (vercelFunctionRegionsVar) {
+  vercelFunctionRegions = vercelFunctionRegionsVar.split(',');
+}
+
+/**
+ * Normalizes the regions config that comes from the Next.js edge functions manifest.
+ * Ensures that config like `home` and `global` are converted to the corresponding Vercel region config.
+ * In the future we'll want to make `home` and `global` part of the Build Output API.
+ * - `home` refers to the regions set in vercel.json or on the Vercel dashboard project config.
+ * - `global` refers to all regions.
+ */
+function normalizeRegions(regions: Regions): string[] {
+  if (typeof regions === 'string') {
+    regions = [regions];
+  }
+
+  const newRegions: string[] = [];
+  for (const region of regions) {
+    // Explicitly mentioned as `home` is one of the explicit values for preferredRegion in Next.js.
+    if (region === 'home' && vercelFunctionRegions) {
+      // Includes the regions from the VERCEL_FUNCTION_REGIONS env var.
+      newRegions.push(...vercelFunctionRegions);
+      continue;
+    }
```

Comment:
If `region === 'home'`, but `vercelFunctionRegions` is falsey, then `"home"` will be pushed into `newRegions` below. Is that to be expected?

If not, then an inner `if` would prevent that:

```suggestion
    if (region === 'home') {
      if (vercelFunctionRegions) {
        // Includes the regions from the VERCEL_FUNCTION_REGIONS env var.
        newRegions.push(...vercelFunctionRegions);
      }
      continue;
    }
```